File: C++/Leetcode/Strings/lengthOfLastWord.cpp
#include <iostream>
#include <vector>

class Solution {
    public:
        int lengthOfLastWord(std::string s){
            int c = 0;
            bool bVal = false;
            for (int i = s.size() - 1; i >= 0; i--){
                std::cout << s[i] << std::endl;
                if (s[i] == ' ' and bVal == false){
                    continue;
                }
                else if (isalpha(s[i])){
                    bVal = true;
                    c++;
                }
                else {
                    return c;
                }
            }
            return c;
        }
};

int main(){
    Solution solution_instance;
    std::string s = "   fly me to the   moon";
    std::cout << solution_instance.lengthOfLastWord(s) << std::endl;
}

File: Python/Leetcode/String/lengthOfLastWord.py
class Solution:
    def lengthOfLastWord(self, s):
        s = s.strip()
        c = 0
        for i in range(len(s)-1, -1, -1):
            if s[i] == ' ':
                return c
            c += 1
        return c
    
    def lengthOfLastWord2(self, s):
        return len(s.split()[-1])
        
if __name__ == '__main__':
    solution_instance = Solution()
    print(solution_instance.lengthOfLastWord("   fly me   to   the moon  "))